### PR TITLE
AN-59 Split pullquote and blockquote styling

### LINKED ISSUE #246: Split block quote and pull quote styling
Define styling for block quotes and pull quotes separately. I also expect the default block quote style to take advantage of content inset, similar font size to body test, slightly darker background color with a left border.

### PR DESCRIPTION
**Pull request by kevinfodness**
* Created a new settings block for blockquote-specific settings.
* Changed behavior and appearance of blockquotes to differentiate them from pullquotes.
* Created a settings migration function to use existing body and pullquote settings to initialize blockquote settings.
* Added reusable functions for handling settings migration in the base Apple_News class.
* Refactored test-apple-news.php to conform to WordPress coding standards and PHP best practices.
* Added unit test coverage in test-apple-news.php to cover all settings migration functions.
* Added unit test coverage for blockquotes.

Fixes #246.

### CODE REVIEW

File: includes/class-apple-news.php [hunk 1]
```diff
@@ -161,32 +239,34 @@ public function migrate_api_settings( $wp_settings ) {
...
-				// Adjust font size down by 2 to match legacy handling.
-				if ( 'size' === $setting ) {
-					$wp_settings[ $caption_setting ] -= 2;
-				}
-			}
+		// Clone and modify font size, if necessary.
+		if ( ! isset( $wp_settings['caption_size'] )
+		     && isset( $wp_settings['body_size'] )
+		) {
+			$wp_settings['caption_size'] = $wp_settings['body_size'] - 2;
```
Comment (mslinnea) [inline includes/class-apple-news.php]: Do you need to take any steps here to ensure that `$wp_settings['body_size']` is an int?

File: includes/class-apple-news.php [hunk 2]
```diff
@@ -211,38 +291,32 @@ public function migrate_header_settings( $wp_settings ) {
...
+			'header3_line_height' => 'header_line_height',
+			'header4_line_height' => 'header_line_height',
+			'header5_line_height' => 'header_line_height',
+			'header6_line_height' => 'header_line_height',
+		) );
+
+		// Remove legacy settings.
+		unset( $wp_settings['header_color'] );
+		unset( $wp_settings['header_font'] );
+		unset( $wp_settings['header_line_height'] );
```
Comment (mslinnea) [inline includes/class-apple-news.php]: Do you need to check that the above keys exist before unsetting them?

File: includes/class-apple-news.php [hunk 3]
```diff
@@ -318,4 +392,47 @@ public function validate_settings( $wp_settings ) {
...
+	 *
+	 * @param array $compare The array to compare against the list of keys.
+	 * @param array $keys The keys to check.
+	 *
+	 * @access private
+	 * @return bool True if all keys exist in the array, false if not.
+	 */
+	private function _all_keys_exist( $compare, $keys ) {
+		return ( count( $keys ) === count(
+			array_intersect_key( $compare, array_combine( $keys, $keys ) ) )
```
Comment (mslinnea) [inline includes/class-apple-news.php]: Neat way of doing this!

**Comment by mslinnea** [comment]
🐴

File: includes/class-apple-news.php (hunk 2 above)
Comment (kevinfodness) [inline includes/class-apple-news.php]: `unset` works whether the array key exists or not, so no need to look before I leap here.

File: includes/class-apple-news.php (hunk 3 above)
Comment (kevinfodness) [inline includes/class-apple-news.php]: Thanks!

File: includes/class-apple-news.php (hunk 1 above)
Comment (kevinfodness) [inline includes/class-apple-news.php]: Not a bad idea to add a check ahead of time to be safe. Thanks for the suggestion.

**Pull request merged**